topics/functions-intro.md

```diff
@@ -0,0 +1,353 @@
+
+To address these needs while avoiding breaking changes to already-established and well-liked ephemeral scripts, Redis v7.0 introduces Redis Functions.
+
+## What are Redis Functions?
+
+Redis functions are an evolutionary step from ephemeral scripting.
+
+Functions provide the same core functionality as scripts but are first-class software artifacts citizens of the database.
+Redis manages functions as an integral part of the database and ensures their availability via data persistence and replication.
+Because functions are part of the database and therefore declared before use, applications aren't required to load them during runtime nor risk aborted transactions.
+An application that uses functions depends only on their APIs rather than on the embedded script logic in the database.
+
+Whereas ephemeral scripts are considered a part of the application's domain, functions extend the database server itself with user-provided logic.
+Every function has a unique user-provided name, making it much easier to call and trace its execution.
+
+The design of Redis Functions also attempts to demarcate between the programming language used for writing functions and their management by the server.
+Lua, the only language interpreter that Redis presently support as an embedded execution engine, is meant to be simple and easy to learn.
+However, the choice of Lua as a language still presents many Redis users with an obstacle. 
+
+The Redis Functions feature makes no assumptions regarding the implementation's language.
+An execution engine that is part of the definition of the function handles running it.
+An engine can theoretically execute functions in any language as long as it respects several rules (such as the ability to terminate an inflight function's execution).
+
+Presently, as noted above, Redis ships with a single embedded [Lua 5.1](/topics/lua-api) engine.
+There are plans to support additional engines in the future.
+Redis functions can use all of Lua's available capabilities to ephemeral scripts,
+with the only exception being the [Redis Lua scripts debugger](/topics/ldb).
+
+Functions also simplify development by enabling code sharing.
+Every function belongs to a single library, and any given library can consist of multiple functions.
+The library's contents are immutable, and selective updates of its functions aren't allowed.
+Instead, libraries are updated as a whole with all of their functions together in one operation.
+This allows calling functions from other functions within the same library.
+
+Functions are intended to support better the use case of maintaining a consistent view for data entities through a logical schema, as mentioned above.
+As such, functions are stored alongside the data itself.
+Functions are also persisted to the AOF file and synchronized from master to replicas.
```

Comment:
maybe these two lines should be expanded to specify that they're automatically propagated to replicas, but not yet to cluster nodes.
but also stress that propagating them to all cluster nodes is an administrative act, and not a responsibility of the client application or the client library's cluster support.

Reply:
i suppose we need some short section about cluster
* mention how to load them into all cluster nodes
* mention that when adding node it is done automatically

and also mention `redis-cli --functoins-rdb` for the ephemeral case.